File: src/test/java/net/bramp/ffmpeg/PreconditionsCheckInvalidNotEmptyTest.java
package net.bramp.ffmpeg;

import java.util.Arrays;
import java.util.List;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

@RunWith(Parameterized.class)
public class PreconditionsCheckInvalidNotEmptyTest {
  @Parameterized.Parameters(name = "{0}")
  public static List<String> data() {
    return Arrays.asList(null, "", "   ", "\n", " \n ");
  }

  private final String input;

  public PreconditionsCheckInvalidNotEmptyTest(String input) {
    this.input = input;
  }

  @Test(expected = IllegalArgumentException.class)
  public void testUri() {
    Preconditions.checkNotNullEmptyOrBlank(input, "test must throw exception");
  }
}

File: src/test/kotlin/net/bramp/ffmpeg/PreconditionsCheckInvalidNotEmptyTest.kt
package net.bramp.ffmpeg

import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class PreconditionsCheckInvalidNotEmptyTest(private val input: String?) {

  @Test(expected = IllegalArgumentException::class)
  fun testUri() {
    Preconditions.checkNotNullEmptyOrBlank(input, "test must throw exception")
  }

  companion object {
    @JvmStatic
    @Parameterized.Parameters(name = "{0}")
    fun data(): List<String?> = listOf(null, "", "   ", "\n", " \n ")
  }
}
